framework/src/main/java/fi/otavanopisto/pyramus/koski/KoskiClient.java

```diff
@@ -269,6 +268,54 @@ public void updatePerson(Person person) throws KoskiException {
     }
   }
 
+  private KoskiStudentHandler getHandlerType(KoskiStudyProgrammeHandler handler) {
+    switch (handler) {
+      case aikuistenperusopetus:
+        return aikuistenPerusopetuksenHandler;
+      case lukio:
+        return lukioHandler;
+      case aineopiskeluperusopetus:
+      case aineopiskelulukio:
+        return internetixHandler;
+      case aikuistenperusopetuksenalkuvaihe:
+        return apaHandler;
+
+      default:
+        logger.severe(String.format("Handler for type %s couldn't be determined.", handler));
+        return null;
+    }
+  }
+
+  class SourceSystemId {
+    public SourceSystemId(KoskiStudyProgrammeHandler handler, Long studentId) {
+      this.handler = handler;
+      this.studentId = studentId;
+    }
+    
+    public Long getStudentId() {
+      return studentId;
+    }
+
+    public KoskiStudyProgrammeHandler getHandler() {
+      return handler;
+    }
+
+    private final KoskiStudyProgrammeHandler handler;
+    private final Long studentId;
+  }
+  
+  private SourceSystemId parseSource(LahdejarjestelmaIdReturnVal lahdejarjestelmaIdReturnVal) {
+    String id = lahdejarjestelmaIdReturnVal.getId();
+    
+    if (StringUtils.isNotBlank(id) && id.contains(":")) {
+      KoskiStudyProgrammeHandler handler = KoskiStudyProgrammeHandler.valueOf(id.substring(0, id.indexOf(':')));
+      Long studentId = Long.valueOf(id.substring(id.indexOf(':') + 1));
```

Comment:
No biggie but could also use StringUtils.substringBefore and StringUtils.substringAfter